src/components/Receiving/ItemDetails.js

```diff
@@ -173,17 +173,24 @@ class ItemDetails extends Component {
   );
 
   onChangeField = (item, value, key) => {
-    this.setState(({ lineItems }) => {
+    this.setState(({ allChecked, lineItems }) => {
       const updatedLineItems = { ...lineItems };
+      const allCheckedItems = { ...allChecked };
+
       const selectedItem = updatedLineItems[item.poLineId].filter(el => el.id === item.id)[0];
 
       selectedItem[key] = value;
+      selectedItem.isSelected = key === 'barcode' && selectedItem[key].length > 0;
+      const isAllChecked = updatedLineItems[item.poLineId].find(line => !line.isSelected || line.isSelected === false);
```

Comment:
@Feboll Have you checked the option of using `toggleItem(item)`?

Reply:
Yes, I checked. I think this solution is simpler, because in the toggleItem we again find the selected Item and call setState again. I also need to check if I chose this item earlier or not. This is correct for toggleAll.

Reply:
I don't think that we need to duplicate item selecting logic. Could you please refactor this place and leverage select item logic in a function, as well as remove `selectedItem.isSelected = key === 'barcode'` from common `onChangeField` ? it  could be `changeBarcode()` callback, for example.

Reply:
agreed with @aliaksei-chumakou